### ISSUE TITLE
DateTime formatter should support the 'gg' token

### ISSUE THREAD

**Issue by sebastienros** [OrchardCMS/Orchard#5595]
Windows 10 comes with some like this, and this makes the unit tests fail on the cultures containing 'gg'.

**Comment by Piedone** [comment]
ggwp?

**Comment by BenedekFarkas** [comment]
Never give up!

**Comment by DanielStolt** [comment]
Didn't I fix this a few month back @sebastienros when you first installed Windows 10?

**Comment by DanielStolt** [comment]
Yes, I did! 9491a575

**Comment by sebastienros** [comment]
Forgot about it, but dev branch :/

**Comment by DanielStolt** [comment]
We could cherry pick it over to 1.9.x I think?

**Comment by sebastienros** [comment]
I tried to cherry pick the changeset, but some tests still fail. The ones named ParseDateTime01 and ParseDateTime02 (at least).

```
Test Name:	ParseDateTimeTest01
Test FullName:	Orchard.Tests.Localization.DefaultDateFormatterTests.ParseDateTimeTest01
Test Source:	C:\Users\sebros\Documents\Projects\Orchard\src\Orchard.Tests\Localization\DefaultDateFormatterTests.cs : line 26
Test Outcome:	Failed
Test Duration:	0:02:57.744

Result StackTrace:	
at Orchard.Tests.Localization.DefaultDateFormatterTests.ParseDateTimeTest01() in C:\Users\sebros\Documents\Projects\Orchard\src\Orchard.Tests\Localization\DefaultDateFormatterTests.cs:line 69
--AssertionException
at Orchard.Tests.Localization.DefaultDateFormatterTests.<>c__DisplayClass1_0.<ParseDateTimeTest01>b__0(CultureInfo culture) in C:\Users\sebros\Documents\Projects\Orchard\src\Orchard.Tests\Localization\DefaultDateFormatterTests.cs:line 59
Result Message:	
System.AggregateException : Parse tests failed for 768 of 239760 cases. Expected 0 failed cases or less.
  ----> NUnit.Framework.AssertionException :   Expected: 1998-6-1 10:30:0.0-Unspecified-00:00:00
  But was:  1998-6-1 22:30:0.0-Unspecified-00:00:00
```

**Comment by DanielStolt** [comment]
Do they pass on dev?

**Comment by jvdonkoec** [comment]
I have, on dev and on 1.9.x, 6 failing unit tests.

1 if this type:
Orchard.Tests.Localization.DefaultDateFormatterTests.ParseDateTest01:
System.AggregateException : Parse tests failed for 4 of 39792 cases. Expected 0 failed cases or less.
  ----> NUnit.Framework.AssertionException :   Expected: 1998-7-1 But was:  1998-2-1

Which fails because of, as far as I can see, duplicating abbreviated month names(kej.) for the yo(Yoruba) culture:

![image](https://cloud.githubusercontent.com/assets/12745867/9173400/30b788e4-3f7a-11e5-8e07-ea2dfed43dbc.png)


And 5 of this type:
Orchard.Tests.Localization.DefaultDateFormatterTests.ParseDateTimeTest01:
System.AggregateException : Parse tests failed for 218 of 206604 cases. Expected 0 failed cases or less.
  ----> System.FormatException : The string was not recognized as a valid time. The hour is in 12-hour notation but no AM/PM designator was found.

Not yet digged into the last one but I wonder If it is a local problem(can't see why though) or if these unit tests are failing on other computers as well.

How can I be of further assistance to this problem? It is preventing me to upgrade to 1.9.

This is not the first locale problem I encountered in Orchard, this one is from me too:

https://orchard.codeplex.com/discussions/541722

**Comment by DanielStolt** [comment]
Ok I'll have a look at these. @jvdonkoec what's your configured culture/locale on your machine?

**Comment by pszmyd** [comment]
@Piedone "ggwp?" Just won a LoL game and rotfled :D

**Comment by jvdonkoec** [comment]
@DanielStolt I tried both English - United States and Dutch - Netherlands. 

Both fail at the same tests.

**Comment by DanielStolt** [comment]
Fixed in 7533cd8. All unit tests now pass for me on Windows 10. @jvdonkoec please retest on 1.9.x and let me know if any tests are still failing on your machine.

**Comment by jvdonkoec** [comment]
@DanielStolt On Windows 10 and VS2015 all tests on both on dev and 1.9.x are running fine now. With the Dutch - Netherlands locale active. 

On Windows 7 and VS2013 however the same Localization tests are still failing(my comment 8 days ago mentioning the 6 failing unit tests ran on this setup). 

So I tried W10 and VS2015 just to see if I saw the same issues with the unit tests, resulting in 5 failing unit tests. The Yoruba culture related(the tripple Kej. abbreviated month name) failing unit test disappeared just by using VS2015 on W10.

The remaining 5 failing unit tests on W10 and VS2015 were fixed with your last commit.

So, looks like I'm upgrading to W10 and VS2015 for Orchard. :) Thank you fixing this issue, at least on W10 and VS2015.
